src/pykeen/models/inductive/base.py

```diff
@@ -0,0 +1,101 @@
+        :param entity_representations_kwargs:
+            additional keyword-based parameters for the training entity representations
+        :param validation_factory:
+            the validation factory
+        :param testing_factory:
+            the testing factory. If None, the validation factory is re-used, i.e., validation and test entities come
+            from the same (unseen) set of entities.
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`ERModel.__init__`
+        """
+        super().__init__(
+            triples_factory=triples_factory,
+            entity_representations=entity_representations,
+            entity_representations_kwargs=entity_representations_kwargs,
+            **kwargs,
+        )
+        # entity representation kwargs may contain a triples factory, which needs to be replaced
+        entity_representations_kwargs = entity_representations_kwargs or {}
+        # entity_representations_kwargs.pop("triples_factory", None)
+        # note: this is *not* a nn.ModuleDict; the modules have to be registered elsewhere
+        self._mode_to_representations = {TRAINING: self.entity_representations}
+        if "triples_factory" in entity_representations_kwargs:
+            entity_representations_kwargs = ChainMap(
+                dict(triples_factory=validation_factory), entity_representations_kwargs
+            )
+        self._mode_to_representations[VALIDATION] = validation_entity_representations = self._build_representations(
+            triples_factory=validation_factory,
+            representations=entity_representations,
+            representations_kwargs=entity_representations_kwargs,
+            label="entity",
+        )
+
+        # shared
+        if testing_factory is None:
+            testing_entity_representations = validation_entity_representations
+        else:
+            # non-shared
+            if "triples_factory" in entity_representations_kwargs:
+                entity_representations_kwargs = ChainMap(
+                    dict(triples_factory=testing_factory), entity_representations_kwargs
+                )
+            testing_entity_representations = self._build_representations(
+                triples_factory=testing_factory,
+                representations=entity_representations,
+                representations_kwargs=entity_representations_kwargs,
+                label="entity",
+            )
+        self._mode_to_representations[TESTING] = testing_entity_representations
+
+    # docstr-coverage: inherited
+    def _get_entity_representations_from_inductive_mode(
+        self, *, mode: Optional[InductiveMode]
+    ) -> Sequence[Representation]:  # noqa: D102
+        if mode in self._mode_to_representations:
+            return self._mode_to_representations[mode]
+        elif mode is None:
+            raise ValueError(f"{self.__class__.__name__} does not support inductive mode: {mode}")
```

Comment:
Typo in error?

Reply:
oh wait I see what's going on here. I think the check for mode being none should be first, then after check if in self._mode_to_representations, then else is saying it doesn't support the given mode

Reply:
The `else` branch seems to be already covered by the type checker - `Optional[InductiveMode]` is equivalent to `None | 'training' | 'validation' | 'testing'`. Thus, everything which is compliant with the type annotation ends up in one of the other two cases. This check is only useful for runtime, if a user would try to put in some other value.

So, I would suggest to get rid of one of the two cases and only differentiate between supported (i.e. a key in `_mode_to_representations`) and non-supported (everything else).

Reply:
https://github.com/pykeen/pykeen/pull/1106/commits/f58ade20dee8b0c42a9a00d6f88b79e2662f37d1

Reply:
Made it a bit more explicit in 2611d3a